### PR TITLE
[Fix #184] Seperate UTF8 from Ansi

### LINKED ISSUE #184: [Bug] Utf8 doesn't work in 4.5.0.3
Before submitting a new issue, please verify and check:

 - [x] The issue is specific to Raylib-cs and not raylib
 - [x] I checked there is no similar issue already reported
 - [x] My code has no errors or misuse of Raylib-cs

### Issue description

Since 4.5.0.3, drawing utf8 text shows wrong characters.

### Environment

I'm on Windows, with last version of Raylib-cs.

### Issue screenshot

4.5.0.2 :
image

4.5.0.3 :
image

### PR DESCRIPTION
**Pull request by n77y**
Fix for #184. The issue was that raylib text drawing functions do not use ansi - rather UTF8.

These changes should fix the issue by converting to the correct format depending on what each util method needs.

### CODE REVIEW

**Review by 9ParsonsB — APPROVED** [review]

**Comment by ChrisDill** [comment]
@n77y Not sure as it is hard to test this. Can you change back to utf8 for all except the different fileName arguments as that is where the original issue occurred. Thanks!

File: Raylib-cs/types/Raylib.Utils.cs [hunk 1]
```diff
@@ -9,14 +9,14 @@ public static unsafe partial class Raylib
         /// <summary>Initialize window and OpenGL context</summary>
         public static void InitWindow(int width, int height, string title)
         {
-            using var str1 = title.ToUTF8Buffer();
+            using var str1 = title.ToAnsiBuffer();
```
Comment (ChrisDill) [inline Raylib-cs/types/Raylib.Utils.cs:12]: Keep as ToUTF8Buffer.

File: Raylib-cs/types/Raylib.Utils.cs [hunk 2]
```diff
@@ -9,14 +9,14 @@ public static unsafe partial class Raylib
...
             InitWindow(width, height, str1.AsPointer());
         }
 
         /// <summary>Set title for window (only PLATFORM_DESKTOP)</summary>
         public static void SetWindowTitle(string title)
         {
-            using var str1 = title.ToUTF8Buffer();
+            using var str1 = title.ToAnsiBuffer();
```
Comment (ChrisDill) [inline Raylib-cs/types/Raylib.Utils.cs:19]: Keep as ToUTF8Buffer.

File: Raylib-cs/types/Raylib.Utils.cs [hunk 3]
```diff
@@ -35,7 +35,7 @@ public static string GetClipboardText_()
         /// <summary>Set clipboard text content</summary>
         public static void SetClipboardText(string text)
         {
-            using var str1 = text.ToUTF8Buffer();
+            using var str1 = text.ToAnsiBuffer();
```
Comment (ChrisDill) [inline Raylib-cs/types/Raylib.Utils.cs:38]: Keep as ToUTF8Buffer.

File: Raylib-cs/types/Raylib.Utils.cs [hunk 4]
```diff
@@ -56,73 +56,73 @@ public static int SetGamepadMappings(string mappings)
...
             return LoadShader(str1.AsPointer(), str2.AsPointer());
         }
 
         /// <summary>Load shader from code string and bind default locations</summary>
         public static Shader LoadShaderFromMemory(string vsCode, string fsCode)
         {
-            using var str1 = vsCode.ToUTF8Buffer();
-            using var str2 = fsCode.ToUTF8Buffer();
+            using var str1 = vsCode.ToAnsiBuffer();
+            using var str2 = fsCode.ToAnsiBuffer();
```
Comment (ChrisDill) [inline Raylib-cs/types/Raylib.Utils.cs:68]: Keep as ToUTF8Buffer.

File: Raylib-cs/types/Raylib.Utils.cs [hunk 5]
```diff
@@ -56,73 +56,73 @@ public static int SetGamepadMappings(string mappings)
...
             return LoadShaderFromMemory(str1.AsPointer(), str2.AsPointer());
         }
 
         /// <summary>Get shader uniform location</summary>
         public static int GetShaderLocation(Shader shader, string uniformName)
         {
-            using var str1 = uniformName.ToUTF8Buffer();
+            using var str1 = uniformName.ToAnsiBuffer();
```
Comment (ChrisDill) [inline Raylib-cs/types/Raylib.Utils.cs:75]: Keep as ToUTF8Buffer.

File: Raylib-cs/types/Raylib.Utils.cs [hunk 6]
```diff
@@ -56,73 +56,73 @@ public static int SetGamepadMappings(string mappings)
...
             return GetShaderLocation(shader, str1.AsPointer());
         }
 
         /// <summary>Get shader attribute location</summary>
         public static int GetShaderLocationAttrib(Shader shader, string attribName)
         {
-            using var str1 = attribName.ToUTF8Buffer();
+            using var str1 = attribName.ToAnsiBuffer();
```
Comment (ChrisDill) [inline Raylib-cs/types/Raylib.Utils.cs:82]: Keep as ToUTF8Buffer.

File: Raylib-cs/Raylib-cs.csproj [hunk 7]
```diff
@@ -53,6 +53,7 @@
     <Compile Include="interop\*.cs" />
     <Compile Include="types\*.cs" />
     <Compile Include="types\native\CBool.cs" />
+	<Compile Include="types\native\AnsiBuffer.cs" />
```
Comment (ChrisDill) [inline Raylib-cs/Raylib-cs.csproj:56]: Fix indentation.

**Review by ChrisDill — CHANGES_REQUESTED** [review]
Added comments to make it clear which to change. I think we can also keep ansi for the fileType as well.

**Review by ChrisDill — APPROVED** [review]

**Comment by ChrisDill** [comment]
Thanks, merged!

**Pull request merged**